File: src/main/java/field/object/drop/ItemDrop.java
package field.object.drop;

import client.Character;
import client.inventory.slots.ItemSlot;
import field.object.FieldObject;
import lombok.NonNull;
import net.maple.packets.CharacterPackets;

public class ItemDrop extends AbstractFieldDrop {

    private final ItemSlot item;

    public ItemDrop(@NonNull byte enterType, @NonNull byte leaveType, @NonNull int owner, @NonNull FieldObject source, ItemSlot item) {
        super(enterType, leaveType, owner, source);
        this.item = item;
    }

    @Override
    public boolean isMeso() {
        return false;
    }

    @Override
    public int getInfo() {
        return item.getTemplateId();
    }

    @Override
    public void pickUp(Character chr) {
        field.leave(this, getLeaveFieldPacket(chr));
        CharacterPackets.modifyInventory(chr, i -> i.add(item), true);
    }
}

File: src/main/java/field/object/drop/ItemDrop.kt
package field.`object`.drop

import client.Character
import client.inventory.slots.ItemSlot
import client.messages.ItemDropPickUpMessage
import field.`object`.FieldObject
import net.maple.packets.CharacterPackets

class ItemDrop(owner: Int, source: FieldObject, private val item: ItemSlot, questId: Int) : AbstractFieldDrop(owner, source, questId) {

    override val isMeso = false
    override val info = item.templateId

    override fun pickUp(chr: Character) {
        leaveType = LeaveType.PICKUP
        field.leave(this, getLeaveFieldPacket(chr))
        CharacterPackets.modifyInventory(chr, { it.add(item) }, true)
        chr.write(CharacterPackets.message(ItemDropPickUpMessage(item.templateId)))
    }
}